### PR TITLE
feat(cb2-6248): amend hgv & trl vehicle summary section

### PR DESCRIPTION
**Pull request by HazelWilderspin**
Amend HGV & TRL Vehicle Summary section

Updating hgv and trl template 
[Jira ticket](https://dvsa.atlassian.net/browse/CB2-6248)

## Checklist

- [x] Branch is rebased against the latest develop/common
- [x] Necessary `id` required prepended with `"test-"` have been checked with automation testers and added
- [x] Code and UI has been tested manually after the additional changes
- [x] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [x] Link to the PR added to the repo
- [ ] Delete branch after merge

### CODE REVIEW

File: src/app/forms/templates/hgv/hgv-tech-record.template.ts [hunk 1]
```diff
@@ -9,121 +15,150 @@ export const HgvTechRecord: FormNode = {
       name: 'vehicleType',
       label: 'Vehicle type',
       value: '',
-
+      width: FormNodeWidth.XS,
       type: FormNodeTypes.CONTROL,
-      viewType: FormNodeViewTypes.VEHICLETYPE
+      viewType: FormNodeViewTypes.VEHICLETYPE,
+      editType: FormNodeEditTypes.SELECT,
```
Comment (aaron-downing) [inline src/app/forms/templates/hgv/hgv-tech-record.template.ts:21]: I believe editing the vehicle type is out of scope as it is a lot more complex

File: src/app/forms/templates/hgv/hgv-tech-record.template.ts [hunk 2]
```diff
@@ -9,121 +15,150 @@ export const HgvTechRecord: FormNode = {
...
     },
     {
       name: 'manufactureYear',
       label: 'Year of manufacture',
       value: '',
-
+      width: FormNodeWidth.XS,
       type: FormNodeTypes.CONTROL,
-      viewType: FormNodeViewTypes.STRING
+      editType: FormNodeEditTypes.TEXT,
```
Comment (aaron-downing) [inline src/app/forms/templates/hgv/hgv-tech-record.template.ts:44]: Not sure whether we need to validate a year?

File: src/app/forms/templates/hgv/hgv-tech-record.template.ts (hunk 1 above)
Comment (HazelWilderspin) [reply src/app/forms/templates/hgv/hgv-tech-record.template.ts:21]: It is, both out of scope and a lot more complex, I just made it pretty, like you suggested I do in other places :o)

**Review by kpearmanbjss — DISMISSED** [review]

File: src/app/forms/templates/hgv/hgv-tech-record.template.ts [hunk 3]
```diff
@@ -9,121 +15,150 @@ export const HgvTechRecord: FormNode = {
...
-              value: '',
-
-              type: FormNodeTypes.CONTROL,
-              viewType: FormNodeViewTypes.STRING
-            }
-          ]
-        }
-      ]
     },
+    // {   /// Not on UI model but may be needed later
```
Comment (Ian-Hodges) [inline src/app/forms/templates/hgv/hgv-tech-record.template.ts:74]: I'd remove all the sections of comment-out code. Especially if we don't know if it will ever be needed.

File: src/app/forms/templates/hgv/hgv-tech-record.template.ts [hunk 4]
```diff
@@ -9,121 +15,150 @@ export const HgvTechRecord: FormNode = {
       name: 'vehicleType',
       label: 'Vehicle type',
       value: '',
-
+      width: FormNodeWidth.XS,
       type: FormNodeTypes.CONTROL,
-      viewType: FormNodeViewTypes.VEHICLETYPE
+      viewType: FormNodeViewTypes.VEHICLETYPE,
```
Comment (tom-bjss) [inline src/app/forms/templates/hgv/hgv-tech-record.template.ts:20]: This is out of scope currently as we dont know how this is going to effect the form

File: src/app/forms/templates/trl/trl-tech-record.template.ts [hunk 5]
```diff
@@ -9,42 +14,55 @@ export const TrlTechRecordTemplate: FormNode = {
       name: 'vehicleType',
       label: 'Vehicle type',
       value: '',
-
+      width: FormNodeWidth.XS,
       type: FormNodeTypes.CONTROL,
-      viewType: FormNodeViewTypes.VEHICLETYPE
+      viewType: FormNodeViewTypes.VEHICLETYPE,
```
Comment (tom-bjss) [inline src/app/forms/templates/trl/trl-tech-record.template.ts:19]: As above out of scope

**Review by tom-bjss — COMMENTED** [review]
Good overall but I wouldnt want to approve this with vehicle type being editable, there are 3 that are out of scope but this is the big one for me, we dont want to just be changing vehicle types now without knowing what thats going to do.

e.g. do we need to create the form again but for the new vehicle and try to transfer shared values ?

File: src/app/forms/templates/hgv/hgv-tech-record.template.ts (hunk 3 above)
Comment (HazelWilderspin) [reply src/app/forms/templates/hgv/hgv-tech-record.template.ts:74]: It will probably be needed as soon as the wider axle management problem is resolved

File: src/app/forms/templates/hgv/hgv-tech-record.template.ts (hunk 2 above)
Comment (tom-bjss) [reply src/app/forms/templates/hgv/hgv-tech-record.template.ts:39]: Dont they want to change this anyway in the etl into nop ?

**Review by tom-bjss — APPROVED** [review]

File: src/app/forms/templates/trl/trl-tech-record.template.ts [hunk 6]
```diff
@@ -9,42 +14,50 @@ export const TrlTechRecordTemplate: FormNode = {
...
-
       type: FormNodeTypes.CONTROL,
-      viewType: FormNodeViewTypes.STRING
+      viewType: FormNodeViewTypes.DATE,
+      editType: FormNodeEditTypes.DATE,
+      validators: [{ name: ValidatorNames.Required }],
+      isoDate: false
     },
+
+    // {
```
Comment (aaron-downing) [inline src/app/forms/templates/trl/trl-tech-record.template.ts:55]: Realise this is uncommented on HGVs, is it meant to not be visible for TRL?

Comment (aaron-downing) [reply src/app/forms/templates/trl/trl-tech-record.template.ts:55]: Just seen the service design :O

**Review by kpearmanbjss — APPROVED** [review]
